### PR TITLE
Tests XML reporter

### PR DESCRIPTION
**Pull request by borkaehw**
Currently the test results all go to `<system-out>`, which we are not getting the benefit of a xml file. This format works great with Jenkins test results

### CODE REVIEW

**Comment by borkaehw** [comment]
@andyscott Friendly ping here. I think this is valuable.

**Comment by jjudd** [comment]
Very glad to see this.

As far as I know, this doesn't work with process level isolation. Anyone have any thoughts on how to get that to work or if it is valuable? If it doesn't work and we accept that, we should probably add a note about it somewhere.

**Comment by jjudd** [comment]
Also - thanks for doing this!

**Comment by borkaehw** [comment]
Thanks @jjudd to mention that. If we merge this PR, we should update the documentation as well.

File: src/main/scala/higherkindness/rules_scala/common/sbt-testing/JUnitXmlReporter.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import sbt.testing.{Event, Status, TestSelector}
+import Status.{Canceled, Error, Failure, Ignored, Pending, Skipped}
+import scala.collection.mutable.ListBuffer
+import scala.xml.{Elem, XML}
+
+class JUnitXmlReporter(tasksAndEvents: ListBuffer[(String, ListBuffer[Event])]) {
+  def result: Elem =
+    <testsuites>
+      {
+        for ((name, events) <- tasksAndEvents) yield <testsuite hostname="" name={
```
Comment (andyscott) [inline src/main/scala/higherkindness/rules_scala/common/sbt-testing/JUnitXmlReporter.scala]: Can we just emit strings instead of using XML literal syntax?

**Comment by andyscott** [comment]
Other than my gripe with the XML syntax usage, 👍 from me for merging.

**Pull request merged**